src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs

```diff
@@ -675,11 +699,6 @@ private bool TryTakeWithNoTimeValidation([MaybeNullWhen(false)] out T item, int
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            //If the collection is completed then there is no need to wait.
-            if (IsCompleted)
```

Comment:
Should be replaced with `Debug.Assert(IsCompleted);`

Reply:
> Should be replaced with `Debug.Assert(IsCompleted);`

IsCompleted can be changed from antoher thread , the assert would not be always true https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-69631-merge-a83f55ae749f443aa3/System.Collections.Concurrent.Tests/1/console.b9f2d628.log?helixlogtype=result